flink-runtime/src/test/java/org/apache/flink/runtime/highavailability/JobResultStoreContractTest.java

```diff
@@ -44,106 +45,140 @@ public interface JobResultStoreContractTest {
+                                .hasCleanJobResultEntryAsync(DUMMY_JOB_RESULT_ENTRY.getJobId())
+                                .get())
+                .isTrue();
+        assertThat(jobResultStore.hasJobResultEntryAsync(DUMMY_JOB_RESULT_ENTRY.getJobId()).get())
                 .isTrue();
-        assertThat(jobResultStore.hasJobResultEntry(DUMMY_JOB_RESULT_ENTRY.getJobId())).isTrue();
     }
 
     @Test
-    default void testHasJobResultEntryWithEmptyStore() throws IOException {
+    default void testHasJobResultEntryWithEmptyStore()
+            throws IOException, ExecutionException, InterruptedException {
         JobResultStore jobResultStore = createJobResultStore();
         JobID jobId = new JobID();
-        assertThat(jobResultStore.hasDirtyJobResultEntry(jobId)).isFalse();
-        assertThat(jobResultStore.hasCleanJobResultEntry(jobId)).isFalse();
-        assertThat(jobResultStore.hasJobResultEntry(jobId)).isFalse();
+        assertThat(jobResultStore.hasDirtyJobResultEntryAsync(jobId).get()).isFalse();
+        assertThat(jobResultStore.hasCleanJobResultEntryAsync(jobId).get()).isFalse();
+        assertThat(jobResultStore.hasJobResultEntryAsync(jobId).get()).isFalse();
     }
 
     @Test
-    default void testGetDirtyResultsWithNoEntry() throws IOException {
+    default void testGetDirtyResultsWithNoEntry()
+            throws IOException, ExecutionException, InterruptedException {
         JobResultStore jobResultStore = createJobResultStore();
-        assertThat(jobResultStore.getDirtyResults()).isEmpty();
+        assertThat(jobResultStore.getDirtyResultsAsync().get()).isEmpty();
     }
 
     @Test
-    default void testGetDirtyResultsWithDirtyEntry() throws IOException {
+    default void testGetDirtyResultsWithDirtyEntry()
+            throws IOException, ExecutionException, InterruptedException {
         JobResultStore jobResultStore = createJobResultStore();
-        jobResultStore.createDirtyResult(DUMMY_JOB_RESULT_ENTRY);
+        jobResultStore.createDirtyResultAsync(DUMMY_JOB_RESULT_ENTRY).get();
         assertThat(
-                        jobResultStore.getDirtyResults().stream()
+                        jobResultStore.getDirtyResultsAsync().get().stream()
                                 .map(JobResult::getJobId)
                                 .collect(Collectors.toList()))
                 .singleElement()
                 .isEqualTo(DUMMY_JOB_RESULT_ENTRY.getJobId());
     }
 
     @Test
-    default void testGetDirtyResultsWithDirtyAndCleanEntry() throws IOException {
+    default void testGetDirtyResultsWithDirtyAndCleanEntry()
+            throws IOException, ExecutionException, InterruptedException {
         JobResultStore jobResultStore = createJobResultStore();
-        jobResultStore.createDirtyResult(DUMMY_JOB_RESULT_ENTRY);
-        jobResultStore.markResultAsClean(DUMMY_JOB_RESULT_ENTRY.getJobId());
+        jobResultStore.createDirtyResultAsync(DUMMY_JOB_RESULT_ENTRY).get();
```

Comment:
one tip: Using `join()` instead of `get()` reduces the diff for test methods because you wouldn't have to specifying `ExecutionException` and `InterruptedException` in the test method's signature.

Reply:
Fixed.